test/utils/metadata.test.js

```diff
@@ -0,0 +1,338 @@
+                await assetMetadata.extract();
+                const metadata = assetMetadata.metadata;
+
+                // exclude command duration from mediainfo
+                delete t.output.commandDuration;
+                delete metadata.commandDuration;
+
+                assert.deepStrictEqual(metadata, t.output);
+            });
+        });
+    });
+
+    context('failure tests', () => {
+        [{
+            name: 'no source',
+            expectedErr: {
+                reason: Reason.GenericError,
+                msg: 'No source provided'
+            }
+        }, {
+            name: 'no name, path, mimetype (cannot get extension)',
+            input: {},
+            expectedErr: {
+                reason: Reason.GenericError,
+                msg: `Provided source must contain one of 'name', 'path', or 'mimetype' fields`
+            }
+        }, {
+            name: 'fail to get extension using only name without extension',
+            input: { name: 'test' },
+            expectedErr: {
+                reason: Reason.GenericError,
+                msg: `Cannot determined asset extension`
+            }
+        }, {
+            name: 'fail to get extension using only unknown mimetype',
+            input: { mimetype: 'test' },
+            expectedErr: {
+                reason: Reason.GenericError,
+                msg: `Cannot determined asset extension`
+            }
+        }, {
+            name: 'fail to get extension using name without extension and unknown mimetype',
+            input: { name: 'test', mimetype: 'test' },
+            expectedErr: {
+                reason: Reason.GenericError,
+                msg: `Cannot determined asset extension`
+            }
+        }, {
+            name: 'valid video asset',
+            input: {
+                name: 'file-hd.mov',
+                path: 'test/files/file-hd.mov'
+            },
+            runExtract: true,
+        }, {
+            name: 'video asset neither url nor path',
+            input: {
+                name: 'file-hd.mov'
+            },
+            runExtract: true,
+            expectedErr: {
+                reason: Reason.GenericError,
+                msg: `Image source must contain either 'url' or 'path' field for retrieving metadata`
+            }
+        }, {
+            name: 'valid image asset',
+            input: {
+                name: 'file.jpg',
+                path: 'test/files/file.jpg'
+            },
+            runExtract: true
+        }, {
+            name: 'image asset without path',
+            input: {
+                name: 'file.jpg',
+                url: 'https://example.com/file.jpg'
+            },
+            runExtract: true,
+            expectedErr: {
+                reason: Reason.GenericError,
+                msg: `Image source must contain 'path' field for retrieving metadata`
+            }
+        }].forEach(t => {
```

Comment:
Rename `t` to something more meaningful like `currentTest`

Reply:
done